server/routes/visitJourney/dateAndTime.ts

```diff
@@ -24,65 +24,54 @@ export default class DateAndTime {
     })
 
     let restrictionChangeMessage = ''
-    let matchingSlot = ''
-    let selectedSlot
+    let matchingSlot
 
+    // first time here on update journey, visitSlot.id will be ''
     if (isUpdate && visitSessionData.visitSlot?.id === '') {
-      selectedSlot = getSelectedSlotByStartTimestamp(
+      matchingSlot = getSlotByStartTimeAndRestriction(
         slotsList,
         visitSessionData.visitSlot.startTimestamp,
         visitSessionData.visitRestriction,
       )
 
-      if (selectedSlot) {
-        matchingSlot = selectedSlot.id
+      if (
+        matchingSlot &&
+        (matchingSlot.availableTables > 0 ||
+          visitSessionData.visitRestriction === visitSessionData.originalVisitSlot.visitRestriction)
```

Comment:
guess this line is very important

Reply:
yeah, selects the same time slot if it has availability **or** if it's what you had before (i.e. it might be full but you have one of the places)